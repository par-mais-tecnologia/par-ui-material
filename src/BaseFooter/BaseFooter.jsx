import React, { PureComponent } from 'react'
import Grid from '@material-ui/core/Grid'
import { withStyles } from '@material-ui/core/styles'

import PropTypes from 'prop-types'

const styles = (theme) => ({
  footer: {
    position: 'fixed',
    bottom: '0px',
    width: ' 100%',
    margin: '0 auto',
    background: theme.styles && theme.styles.bgFooter ? theme.styles.bgFooter : '#000',
    zIndex: ' 100'
  }
})

class BaseFooter extends PureComponent {
  render () {
    const { children, classes, height } = this.props

    return (
      <Grid container justify='center' style={{ height }} className={classes.footer}>
        {children}
      </Grid>
    )
  }
}

BaseFooter.propTypes = {
  children: PropTypes.oneOfType([
    PropTypes.arrayOf(PropTypes.node),
    PropTypes.node
  ]).isRequired,
  height: PropTypes.number.isRequired
}

export default withStyles(styles)(BaseFooter)
