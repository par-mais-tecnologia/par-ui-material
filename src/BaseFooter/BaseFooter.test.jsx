import React from 'react'
import { configure, shallow } from 'enzyme'
import Adapter from 'enzyme-adapter-react-16'

import BaseFooter from './'

configure({ adapter: new Adapter() })

const getShallow = () => (
  <BaseFooter>
    <h1 className='roboto-light white'>
      Footer content here
    </h1>
  </BaseFooter>
)

describe('baseFooter', () => {
  const wrapper = shallow(getShallow())

  it('shallow BaseFooter', () => {
    expect(wrapper).toBeTruthy()
  })

  it('shallow BaseFooter match Snapshot', () => {
    expect(wrapper).toMatchSnapshot()
  })
})
