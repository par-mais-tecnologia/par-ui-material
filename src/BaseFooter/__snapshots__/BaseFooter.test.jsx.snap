// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`baseFooter shallow BaseFooter match Snapshot 1`] = `
ShallowWrapper {
  Symbol(enzyme.__root__): [Circular],
  Symbol(enzyme.__unrendered__): <WithStyles(BaseFooter)
    height={200}
  >
    <h1
      className="roboto-light white"
    >
      Footer content here
    </h1>
  </WithStyles(BaseFooter)>,
  Symbol(enzyme.__renderer__): Object {
    "batchedUpdates": [Function],
    "getNode": [Function],
    "render": [Function],
    "simulateError": [Function],
    "simulateEvent": [Function],
    "unmount": [Function],
  },
  Symbol(enzyme.__node__): Object {
    "instance": null,
    "key": undefined,
    "nodeType": "class",
    "props": Object {
      "children": <h1
        className="roboto-light white"
      >
        Footer content here
      </h1>,
      "classes": Object {
        "footer": "BaseFooter-footer-1",
      },
      "height": 200,
    },
    "ref": null,
    "rendered": Object {
      "instance": null,
      "key": undefined,
      "nodeType": "host",
      "props": Object {
        "children": "Footer content here",
        "className": "roboto-light white",
      },
      "ref": null,
      "rendered": "Footer content here",
      "type": "h1",
    },
    "type": [Function],
  },
  Symbol(enzyme.__nodes__): Array [
    Object {
      "instance": null,
      "key": undefined,
      "nodeType": "class",
      "props": Object {
        "children": <h1
          className="roboto-light white"
        >
          Footer content here
        </h1>,
        "classes": Object {
          "footer": "BaseFooter-footer-1",
        },
        "height": 200,
      },
      "ref": null,
      "rendered": Object {
        "instance": null,
        "key": undefined,
        "nodeType": "host",
        "props": Object {
          "children": "Footer content here",
          "className": "roboto-light white",
        },
        "ref": null,
        "rendered": "Footer content here",
        "type": "h1",
      },
      "type": [Function],
    },
  ],
  Symbol(enzyme.__options__): Object {
    "adapter": ReactSixteenAdapter {
      "options": Object {
        "enableComponentDidUpdateOnSetState": true,
        "lifecycles": Object {
          "componentDidUpdate": Object {
            "onSetState": true,
          },
          "getDerivedStateFromProps": true,
          "getSnapshotBeforeUpdate": true,
          "setState": Object {
            "skipsComponentDidUpdateOnNullish": true,
          },
        },
      },
    },
  },
}
`;
